packages/contracts-ops/contracts/scripts/UpgradeCallBatches.s.sol

```diff
@@ -0,0 +1,236 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity 0.7.6;
+pragma abicoder v2;
+
+/*//////////////////////////////////////////////////////////////
+                                 IMPORTS
+    //////////////////////////////////////////////////////////////*/
+
+// Utilities
+import {Test} from "forge-std/Test.sol";
+import {console2} from "forge-std/console2.sol";
+
+// Ops libs
+import {Config} from "../Config.sol";
+import {CallBatch} from "../CallBatch.sol";
+
+contract UpgradeCallBatches is Test, Config, CallBatch {
+    /*//////////////////////////////////////////////////////////////
+                                 BEACONS
+    //////////////////////////////////////////////////////////////*/
+
+    address homeBeacon;
+    address replicaBeacon;
+    address governanceRouterBeacon;
+    address bridgeRouterBeacon;
+    address tokenRegistryBeacon;
+    address bridgeTokenBeacon;
+
+    /*//////////////////////////////////////////////////////////////
+                             IMPLEMENTATIONS
+    //////////////////////////////////////////////////////////////*/
+
+    address homeImpl;
+    address replicaImpl;
+    address governanceRouterImpl;
+    address bridgeTokenImpl;
+    address tokenRegistryImpl;
+    address bridgeRouterImpl;
+
+    /*//////////////////////////////////////////////////////////////
+                            GOVERNANCE CALLS
+    //////////////////////////////////////////////////////////////*/
+
+    bytes upgradeHome;
+    bytes upgradeReplica;
+    bytes upgradeGovRouter;
+    bytes upgradeBridgeRouter;
+    bytes upgradeTokenRegistry;
+    bytes upgradeBridgeToken;
+
+    address beaconController;
+    address governanceRouterProxy;
+
+    bytes executeCallBatchCall;
+
+    string configFile;
+    string[] domainNames;
+    string[] networksArray;
+
+    function run(
+        string memory _configFile,
+        string[] memory _domainNames,
+        string memory _localDomainName,
+        bool recovery
+    ) external {
+        localDomainName = _localDomainName;
+        domainNames = _domainNames;
+        configFile = _configFile;
+        setUp();
+        for (uint256 i; i < domainNames.length; i++) {
+            string memory domain = domainNames[i];
+            loadBeacons(domain);
+            loadImplementations(domain);
+            loadController(domain);
+            generateGovernanceCalls(domain);
+        }
+        writeCallBatch(recovery);
+    }
+
+    function setUp() internal {
+        __Config_initialize(configFile);
+        string memory outputFile = "upgradeActions.json";
+        __CallBatch_initialize(
+            localDomainName,
+            getDomainNumber(localDomainName),
+            outputFile,
+            true
+        );
+        networksArray = getNetworks();
+    }
+
+    function loadBeacons(string memory domain) internal {
```

Comment:
why is copying to local state preferable to reading from the config via getters?

i.e. why are we doing 


```
address a;

load() {
   a = a();
}

do_thing() {
   a;
}
```

instead of 
```
do_thing() {
  a()
}
```

Reply:
agree it's cleaner to read from config imo

if we're doing multiple scripts at a time i don't want stale local state from a diff chain causing issues (also more stateful for the developer to think about while reading and editing the code, which is not preferable)